invokeai/app/invocations/metadata.py

```diff
@@ -16,116 +19,99 @@
 
+    def invoke(self, context: InvocationContext) -> MetadataItemOutput:
+        return MetadataItemOutput(item=MetadataItemField(label=self.label, value=self.value))
 
-class ImageMetadata(BaseModelExcludeNull):
-    """An image's generation metadata"""
 
-    metadata: Optional[dict] = Field(
-        default=None,
-        description="The image's core metadata, if it was created in the Linear or Canvas UI",
+@invocation_output("metadata_output")
+class MetadataOutput(BaseInvocationOutput):
+    metadata: MetadataField = OutputField(description="Metadata Dict")
+
+
+@invocation("metadata", title="Metadata", tags=["metadata"], category="metadata", version="1.0.0")
+class MetadataInvocation(BaseInvocation):
+    """Takes a MetadataItem or collection of MetadataItems and outputs a MetadataDict."""
+
+    items: Union[list[MetadataItemField], MetadataItemField] = InputField(
+        description=FieldDescriptions.metadata_item_polymorphic
     )
-    graph: Optional[dict] = Field(default=None, description="The graph that created the image")
 
+    def invoke(self, context: InvocationContext) -> MetadataOutput:
+        if isinstance(self.items, MetadataItemField):
+            # single metadata item
+            data = {self.items.label: self.items.value}
+        else:
+            # collection of metadata items
+            data = {item.label: item.value for item in self.items}
 
-@invocation_output("metadata_accumulator_output")
-class MetadataAccumulatorOutput(BaseInvocationOutput):
-    """The output of the MetadataAccumulator node"""
+        # add app version
+        data.update({"app_version": __version__})
+        return MetadataOutput(metadata=MetadataField.model_validate(data))
 
-    metadata: CoreMetadata = OutputField(description="The core metadata for the image")
 
+@invocation("merge_metadata", title="Metadata Merge", tags=["metadata"], category="metadata", version="1.0.0")
+class MergeMetadataInvocation(BaseInvocation):
+    """Merged a collection of MetadataDict into a single MetadataDict."""
 
-@invocation(
-    "metadata_accumulator", title="Metadata Accumulator", tags=["metadata"], category="metadata", version="1.0.0"
-)
-class MetadataAccumulatorInvocation(BaseInvocation):
-    """Outputs a Core Metadata Object"""
+    collection: list[MetadataField] = InputField(description=FieldDescriptions.metadata_collection)
+
+    def invoke(self, context: InvocationContext) -> MetadataOutput:
+        data = {}
+        for item in self.collection:
+            data.update(item.model_dump())
 
-    generation_mode: Optional[str] = InputField(
+        return MetadataOutput(metadata=MetadataField.model_validate(data))
+
+
+@invocation("core_metadata", title="Core Metadata", tags=["metadata"], category="metadata", version="1.0.0")
+class CoreMetadataInvocation(BaseInvocation):
+    """Collects core generation metadata into a MetadataField"""
+
+    generation_mode: Literal["txt2img", "img2img", "inpaint", "outpaint"] = InputField(
```

Comment:
This line of code breaks SDXL integration: https://github.com/invoke-ai/InvokeAI/issues/4959

It appears to be missing the `sdxl_`-prefixed values.